app/javascript/views/walletServices/BraveConnection.tsx

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import * as React from "react";
+import { FormattedMessage, injectIntl } from "react-intl";
+import routes from "../routes";
+
+import { LoaderIcon } from "brave-ui/components/icons";
+import NotConnected from "./braveConnection/NotConnected";
+import UpholdConnection from "./braveConnection/UpholdConnection";
+import GeminiConnection from "./braveConnection/GeminiConnection";
+
+// This class serves as the entry point for establishing a wallet connection.
+// It allows users to establish a connection to different crypto wallet providers.
+// Upon load makes a request to
+class BraveConnection extends React.Component<any, any> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      geminiConnection: {},
+      isLoading: true,
+      upholdConnection: {},
+    };
+  }
+
+  public componentDidMount() {
+    this.loadData();
+  }
+
+  public render() {
+    // Show the loading component first
+    if (this.state.isLoading) {
+      return <LoadingComponent />;
+      // If there's an uphold connection let's show the UpholdConnection component
+    } else if (
+      this.state.upholdConnection &&
+      this.state.upholdConnection.uphold_id
+    ) {
+      return (
+        <UpholdConnection
+          defaultCurrency={this.state.upholdConnection.default_currency}
+          upholdUsername={this.state.upholdConnection.username}
+          is_payable={this.state.geminiConnection['payable?']}
+          loadData={this.loadData}
+        />
+      );
+      // If there's an gemini connection let's show the GeminiConnection component
+    } else if (
+      this.state.geminiConnection &&
+      this.state.geminiConnection.id
+    ) {
+      return (
+        <GeminiConnection
+          defaultCurrency={this.state.geminiConnection.default_currency}
+          displayName={this.state.geminiConnection.display_name}
+          status={this.state.geminiConnection.uphold_status}
+          loadData={this.loadData}
+        />
+      );
+      // Finally if there was no wallets connected we should give the user the ability to connect.
+    } else {
+      return <NotConnected />;
+    }
+  }
+
+  // Sets loading to true and makes a request to the wallet path in PublishersController#wallet
+  private loadData = () => {
+    this.setState({ isLoading: true });
+
+    axios.get(routes.publishers.wallet.path).then((response) => {
```

Comment:
I sometimes like to use the built in fetch API and save an import 

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

Reply:
Good suggestion! I usually prefer native APIs and we have used `fetch` quite extensively but for these components I wanted some nicer error handling. It's also nice to set defaults for our `Content-Type` and `CRSF token` so we have less verbose code. 🎉 

 https://github.com/brave-intl/publishers/blob/26f23ebe892ad21886c421d34bc07b1eeb09fee5/app/javascript/packs/WalletServices.jsx#L14-L15